src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceMarginViewMargin.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Formatting;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.InheritanceMargin
+{
+    internal class InheritanceMarginViewMargin : IWpfTextViewMargin
+    {
+        private readonly IWpfTextViewHost _textViewHost;
+        private readonly IWpfTextView _textView;
+        private readonly ITagAggregator<InheritanceMarginTag> _tagAggregator;
+        private readonly IOptionService _optionService;
+        private readonly InheritanceGlyphManager _glyphManager;
+        private readonly string _languageName;
+        private bool _refreshAllGlyphs;
+        private bool _disposed;
+        private readonly Grid _grid;
+        private readonly Canvas _mainCanvas;
+
+        // Same size as the Glyph Margin
+        private const double HeightAndWidthOfMargin = 17;
+
+        public InheritanceMarginViewMargin(
+            IWpfTextViewHost textViewHost,
+            IThreadingContext threadingContext,
+            IStreamingFindUsagesPresenter streamingFindUsagesPresenter,
+            IUIThreadOperationExecutor operationExecutor,
+            IClassificationFormatMap classificationFormatMap,
+            ClassificationTypeMap classificationTypeMap,
+            IEditorFormatMap editorFormatMap,
+            IAsynchronousOperationListener listener,
+            ITagAggregator<InheritanceMarginTag> tagAggregator,
+            Document document)
```

Comment:
see comment below.  dont' pass in document here.  too easy to accidentally capture an old SOlution snapshot.